xcode/Subconscious/Shared/Components/Common/MarkupTextViewRepresentable.swift

```diff
@@ -243,6 +230,39 @@ struct MarkupTextViewRepresentable: UIViewRepresentable {
                 )
             )
         }
+        
+        // MARK: - NSTextLayoutManagerDelegate
+                                
+        func textLayoutManager(_ textLayoutManager: NSTextLayoutManager,
+                               textLayoutFragmentFor location: NSTextLocation,
+                               in textElement: NSTextElement) -> NSTextLayoutFragment {
+            
+            // TODO: might be better to hold a ref to textContentStorage somewhere
+            // textContentStorage is a concrete implementation of textContentManager
+            let textContentStorage = textLayoutManager.textContentManager as! NSTextContentStorage
```

Comment:
Same as before, might be nice to have an explicit reference to `textContentStorage` in scope instead of this

Reply:
Yes, Let's avoid `!`, since they crash the app. Better to have a guard clause and no-op + `logger.warning` if we can't unwwrap.

```swift
guard let textContentStorage = self.textLayoutManager.textContentManager as? NSTextContentStorage else {
    MarkupTextViewRepresentable.logger.warning("Could not access textContentStorage")
    return
}
```